Automattic/wp-calypso
# Login: Fix podcasting signup flow

#### Changes proposed in this Pull Request

* Add podcast params via `signup_flow` instead of `signup_url`
* Add `isAnchorFmSignup` check to `getSignupUrl`
* Add `isAnchorFmSignup` prop to `login-links.jsx` and `login-form.jsx` to pass to `getSignupUrl`

#### Testing instructions



<img width="577" alt="Screen Shot 2021-02-19 at 4 23 04 PM" src="https://user-images.githubusercontent.com/1689238/108563672-cee76500-72cf-11eb-8f42-2233db059b36.png">

* Go to 'https://calypso.localhost:3000/new?anchor_podcast=3e51070&anchor_episode=e5f332f1-a347-40d9-a79b-ce30884057e5&spotify_url=https://open.spotify.com/show/6HTZdaDHjqXKDE4acYffoD?si=EVfDYETjQCu7pasVG5D73Q'
* Click "Log in"
* Try logging in with an invalid email, and click "Create a new account" in the error that appears. You should see the Anchor-ified Gutenboarding signup modal.
* Go back to "Log in" and click "Create a new account" in the footer. You should see the Anchor-ified Gutenboarding signup modal.
* Regression: test that the regular Gutenboarding signup links still work correctly.



Related to 455-gh-Automattic/dotcom-manage
